src/benchmarklib/benchmark_table_encoder.cpp

```diff
@@ -0,0 +1,143 @@
+#include "benchmark_table_encoder.hpp"
+
+#include "constant_mappings.hpp"
+#include "encoding_config.hpp"
+#include "resolve_type.hpp"
+#include "storage/base_encoded_segment.hpp"
+#include "storage/base_value_segment.hpp"
+#include "storage/chunk_encoder.hpp"
+#include "storage/table.hpp"
+#include "types.hpp"
+
+namespace {
+
+using namespace opossum;  // NOLINT
+
+SegmentEncodingSpec generate_segment_encoding_spec(const BaseValueSegment&) { return {EncodingType::Unencoded}; }
+
+SegmentEncodingSpec generate_segment_encoding_spec(const ReferenceSegment&) {
+  Fail("Did not expect a ReferenceSegment in base table");
+}
+
+SegmentEncodingSpec generate_segment_encoding_spec(const BaseEncodedSegment& base_encoded_segment) {
+  if (base_encoded_segment.compressed_vector_type()) {
+    switch (*base_encoded_segment.compressed_vector_type()) {
+      case CompressedVectorType::FixedSize1ByteAligned:
+      case CompressedVectorType::FixedSize2ByteAligned:
+      case CompressedVectorType::FixedSize4ByteAligned:
+        return {base_encoded_segment.encoding_type(), VectorCompressionType::FixedSizeByteAligned};
+      case CompressedVectorType::SimdBp128:
+        return {base_encoded_segment.encoding_type(), VectorCompressionType::SimdBp128};
+    }
+
+    Fail("GCC thinks this is reachable");
+  } else {
+    return {base_encoded_segment.encoding_type()};
+  }
+}
+
+ChunkEncodingSpec generate_chunk_encoding_spec(const Chunk& chunk) {
+  auto chunk_encoding_spec = ChunkEncodingSpec{chunk.column_count()};
+
+  for (auto column_id = ColumnID{0}; column_id < chunk.column_count(); ++column_id) {
+    const auto& base_segment = *chunk.get_segment(column_id);
+
+    resolve_data_and_segment_type(base_segment, [&](const auto /* data_type_t */, const auto& segment) {
+      chunk_encoding_spec[column_id] = generate_segment_encoding_spec(segment);
+    });
+  }
+
+  return chunk_encoding_spec;
+}
+
+bool is_chunk_encoding_spec_satisfied(const ChunkEncodingSpec& expected_chunk_encoding_spec,
```

Comment:
Satisfied? Isn't this simply `operator==`?

Reply:
Wish it was... the ChunkEncodingSpec might not specify (i.e. std::nullopt) a vector compression scheme for a, e.g., DictionarySegment - but one will be used regardless. So  a expected ChunkEncodingSpec is satisfied with any ChunkEncodingSpec if it didn't explicitly request a VectorCompression. A bit unintuitive, but that's how the stuff in storage/ is build, not easily refactored either.

I had hoped this comment explains it:

```
    // If an explicit VectorCompressionType is requested, check whether it is used in the Segment. Otherwise, do not
    // care about the VectorCompressionType used.
```

Reply:
ok